pkg/i2gw/providers/apisix/apisix.go

```diff
@@ -50,14 +50,11 @@ func NewProvider(conf *i2gw.ProviderConf) i2gw.Provider {
 	}
 }
 
-// ToIR converts stored Apisix API entities to intermediate.IR
+// ToIR converts stored Apisix API entities to emitter_intermediate.IR
 // including the apisix specific features.
-func (p *Provider) ToIR() (intermediate.IR, field.ErrorList) {
-	return p.resourcesToIRConverter.convertToIR(p.storage)
-}
-
-func (p *Provider) ToGatewayResources(ir intermediate.IR) (i2gw.GatewayResources, field.ErrorList) {
-	return common.ToGatewayResources(ir)
+func (p *Provider) ToIR() (emitter_intermediate.IR, field.ErrorList) {
+	ir, errs := p.resourcesToIRConverter.convertToIR(p.storage)
+	return provider_intermediate.ToEmitterIR(ir), errs
```

Comment:
In reality, providers should not rely on `ToEmitterIR`. That is, `p.resourcesToIRConverter.convertToIR` should just return `EmitterIR`, but I added this function to minimize the code change.